### PR TITLE
Add more plugin architecture documentation

### LINKED ISSUE #20: There needs to be more docs about the plugin architecture

### PR DESCRIPTION
**Pull request by lhorie**
Resolves #20

### CODE REVIEW

File: docs/guides/configuring-plugins.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+```js
+// src/main.js
+import App from 'fusion-react';
+import JWTSession from 'fusion-plugin-jwt';
+import sessionConfig from './config/session';
+
+export default () => {
+  const app = new App();
+  const Session = app.plugin(JWTSession, sessionConfig);
```
Comment (dennisgl) [inline docs/guides/configuring-plugins.md]: I feel this example is confusing as if we want to hide deps away to files. 
```js
import FooPlugin from './plugins/foo-plugin';
import fooPluginConfig from './config/foo-plugin';

export default () => {
  const app = new App();
  const Foo = app.plugin(FooPlugin, {Dep1, Dep2, /*...*/, config: fooPluginConfig});
```

File: docs/guides/creating-endpoints.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  }
+  return next();
+}
+```
+
+There's one issue left with the code above: FusionJS code runs isomorphically by default, but we only want to run that code in the server. To do so, simply add a [code fence](./universal-code.md):
+
+```js
+export default ({}) => {
+  if (_NODE__) {
```
Comment (dennisgl) [inline docs/guides/creating-endpoints.md]: typo -> `__NODE__`

File: docs/guides/creating-providers.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  }
+}
+GreetingProvider.childContextTypes = {
+  ...(GreetingProvider.childContextTypes || {}),
+  greet: PropTypes.function.isRequired,
+};
+
+export default () => (ctx, next) => {
+  if (ctx.element) {
+    ctx.element = <MyProvider>{ctx.element}</MyProvider>;
```
Comment (dennisgl) [inline docs/guides/creating-providers.md]: `<GreetingProvider>`?

**Comment by lhorie** [comment]
!merge

**Pull request merged**